### ISSUE TITLE
Aqara double rocker wall mount switch slow response

### ISSUE THREAD

**Issue by r2ixuz** [ebaauw/homebridge-hue#568]
Hello,

maybe I'm wrong here but I have a problem: I use deconz (with a ConBee II stick) for a couple of weeks without any problems. One Philips Hue light, some Aqara sensors, one Innr SP120 and three Aqara double rocker wall mount switch. With these switches I could normally instantaneously activate my Philips lamp or other things. I use only homebridge and trigger everything with Homekit.

About 5 days ago I installed a new OSRAM LIGHTIFY Flex LED stripe light. It works OK but since then, everytime I press an Aqara switch button it takes couple of seconds or even minutes to activate/trigger the linked Homekit device and/or Homekit scene. The strange thing is, homebridge sees the button press in its logging but no action is taken for a certain amount of time. Sometimes the switches triggers everything at normal speed and 5 hours later it won't do it again.

I did restart the deconz docker container, homebridge container, took the battery out of one of the three double rocker, powered off the new Osram light with zero difference. I even deleted the button action in Homekit with no success.

Anyone any idea how I can get these switches to work instantaneously again like on the first day of use. This was awesome like a real oldschool wallmount switch.

Thanks!

**Comment by ebaauw** [comment]
If you run `homebridge -D`, homebridge-hue logs all requests towards deCONZ.  With that info you should be able to determine if the delay is cause by HomeKit (action is executed late) or by deCONZ (action is executed timely, but deCONZ takes too long to react).

**Comment by r2ixuz** [comment]
OK, I tried that. In this case, I single pressed one rocker so in Homekit an AVM switchtig outlet (Fritzbox dect) and one Osram lamp (deconz) should turn on. This time it took 30 seconds.

Looks like Homekit does strange things? But why? If I activate these devices via Home app they turn on instantly. But not via Xiaomi Aqara rocker... All this started after integrating a new Osram LED stripe via deconz and homebridge-hue plugin into Homekit.

([deconz] is your homebridge-hue plugin)

```
[12/1/2019, 10:04:01 PM] [deconz] PC-Schalter Right: homekit button single press
[12/1/2019, 10:04:01 PM] [deconz] Consumption 6: state changed event: {"consumption":3370,"lastupdated":"2019-12-01T21:04:01"}
[12/1/2019, 10:04:02 PM] [deconz] Phoscon-GW: gateway request 254: get /sensors
[12/1/2019, 10:04:02 PM] [deconz] Phoscon-GW: gateway request 254: ok
[12/1/2019, 10:04:02 PM] [deconz] Phoscon-GW: gateway request 255: get /lights
[12/1/2019, 10:04:02 PM] [deconz] Phoscon-GW: gateway request 255: ok
[12/1/2019, 10:04:07 PM] [deconz] Consumption 6: state changed event: {"consumption":3370,"lastupdated":"2019-12-01T21:04:07"}
[12/1/2019, 10:04:07 PM] [deconz] Phoscon-GW: gateway request 256: get /sensors
[12/1/2019, 10:04:07 PM] [deconz] Phoscon-GW: gateway request 256: ok
[12/1/2019, 10:04:07 PM] [deconz] Phoscon-GW: gateway request 257: get /lights
[12/1/2019, 10:04:07 PM] [deconz] Phoscon-GW: gateway request 257: ok
[12/1/2019, 10:04:11 PM] [deconz] Consumption 6: state changed event: {"consumption":3370,"lastupdated":"2019-12-01T21:04:11"}
[12/1/2019, 10:04:12 PM] [deconz] Phoscon-GW: gateway request 258: get /sensors
[12/1/2019, 10:04:12 PM] [deconz] Phoscon-GW: gateway request 258: ok
[12/1/2019, 10:04:12 PM] [deconz] Phoscon-GW: gateway request 259: get /lights
[12/1/2019, 10:04:12 PM] [deconz] Phoscon-GW: gateway request 259: ok
[12/1/2019, 10:04:17 PM] [deconz] Consumption 6: state changed event: {"consumption":3370,"lastupdated":"2019-12-01T21:04:17"}
[12/1/2019, 10:04:17 PM] [deconz] Phoscon-GW: gateway request 260: get /sensors
[12/1/2019, 10:04:17 PM] [deconz] Phoscon-GW: gateway request 260: ok
[12/1/2019, 10:04:17 PM] [deconz] Phoscon-GW: gateway request 261: get /lights
[12/1/2019, 10:04:17 PM] [deconz] Phoscon-GW: gateway request 261: ok
[12/1/2019, 10:04:22 PM] [deconz] Phoscon-GW: gateway request 262: get /sensors
[12/1/2019, 10:04:22 PM] [deconz] Consumption 6: state changed event: {"consumption":3370,"lastupdated":"2019-12-01T21:04:22"}
[12/1/2019, 10:04:22 PM] [deconz] Phoscon-GW: gateway request 262: ok
[12/1/2019, 10:04:22 PM] [deconz] Phoscon-GW: gateway request 263: get /lights
[12/1/2019, 10:04:22 PM] [deconz] Phoscon-GW: gateway request 263: ok
[12/1/2019, 10:04:27 PM] [deconz] Phoscon-GW: gateway request 264: get /sensors
[12/1/2019, 10:04:27 PM] [deconz] Phoscon-GW: gateway request 264: ok
[12/1/2019, 10:04:27 PM] [deconz] Phoscon-GW: gateway request 265: get /lights
[12/1/2019, 10:04:27 PM] [deconz] Phoscon-GW: gateway request 265: ok
[12/1/2019, 10:04:28 PM] [deconz] Consumption 6: state changed event: {"consumption":3370,"lastupdated":"2019-12-01T21:04:28"}
[12/1/2019, 10:04:32 PM] [deconz] Phoscon-GW: gateway request 266: get /sensors
[12/1/2019, 10:04:32 PM] [deconz] Phoscon-GW: gateway request 266: ok
[12/1/2019, 10:04:32 PM] [deconz] Phoscon-GW: gateway request 267: get /lights
[12/1/2019, 10:04:32 PM] [deconz] Phoscon-GW: gateway request 267: ok
[12/1/2019, 10:04:33 PM] [deconz] Consumption 6: state changed event: {"consumption":3370,"lastupdated":"2019-12-01T21:04:33"}
[12/1/2019, 10:04:35 PM] [deconz] LEDstripe01: homekit on changed from 0 to 1
[12/1/2019, 10:04:35 PM] [FRITZ!Box] Switching outlet 116300090795 to true
[12/1/2019, 10:04:35 PM] [FRITZ!Box] 1 pending api calls
[12/1/2019, 10:04:35 PM] [deconz] Phoscon-GW: gateway request 268: put /lights/3/state {"on":true}
[12/1/2019, 10:04:35 PM] [deconz] Phoscon-GW: gateway request 268: ok
[12/1/2019, 10:04:35 PM] [deconz] LEDstripe01: state changed event: {"alert":null,"bri":3,"colormode":"xy","ct":370,"hue":0,"on":true,"reachable":true,"sat":0,"xy":[0.1739,0.0419]}
```

**Comment by ebaauw** [comment]
```
[12/1/2019, 10:04:01 PM] [deconz] PC-Schalter Right: homekit button single press
```
So here you pressed the switch button
```
[12/1/2019, 10:04:35 PM] [deconz] LEDstripe01: homekit on changed from 0 to 1
```
Here, the HomeKit action updates the light (in HomeKit), indeed over half a minute later:
```
[12/1/2019, 10:04:35 PM] [deconz] Phoscon-GW: gateway request 268: put /lights/3/state {"on":true}
[12/1/2019, 10:04:35 PM] [deconz] Phoscon-GW: gateway request 268: ok
[12/1/2019, 10:04:35 PM] [deconz] LEDstripe01: state changed event: {"alert":null,"bri":3,"colormode":"xy","ct":370,"hue":0,"on":true,"reachable":true,"sat":0,"xy":[0.1739,0.0419]}
```
Homebridge-hue reacts within the same second, sending the update to deCONZ.  The homebridge-hue/deCONZ side would seem fine.

```
[12/1/2019, 10:04:35 PM] [FRITZ!Box] Switching outlet 116300090795 to true
[12/1/2019, 10:04:35 PM] [FRITZ!Box] 1 pending api calls
```
I don't know the FRITZ!Box plugin, in particular if these messages appear before or after the plugin tries to contact the FRITZ!Box.  Could it be blocking HomeKit?  Could you humour me, remove the action for the FRITZ!Box and see how quickly HomeKit reacts in that case?  

> If I activate these devices via Home app they turn on instantly. But not via Xiaomi Aqara rocker...

Do you use a HomeKit scene to set both the light and the outlet, or do you specify both on the button action?

Note that when interacting with the Home app, your iDevice communicates with homebridge.  In case you use the switch, your home hub (AppleTV, iPad, HomePod) does the honours.  Could it be having issues reaching homebridge?

**Comment by r2ixuz** [comment]
No, the other Aqara switches do not switch AVM devices and I have buttons with only scenes and buttons with only "turn lamp on". And all of them have the same issue. I think you are right, that something is wrong with my Apple TV.

Googled something which seems pretty much the same behaviour: https://www.reddit.com/r/HomeKit/comments/7k0fe1/homekit_does_not_fire_actions/ .

I observed the same: If I change something in the Home app for the switches (other lamp to turn on or whatever), this switch works after a couple of turn on/off cycles quick again as intended. Now some hours of not pushing any buttons go by and here you go: the buttons action is invoked hardly delayed or sometimes it even won't work till you go up to the Home app and reconfigure the wall switch.

I think I try next to remove and readd the Apple TV 4K as home hub. Some in the reddit thread mentioned this.. ;-(

**Comment by r2ixuz** [comment]
Ok I guess I found the problem. My iPad was still as standby device for Homekit configured but bluetooth is always turned off there and background app refresh also. This was never really a problem, because my Apple TV was configured to not use sleep mode.

Roughly at the same time when I installed the new Osram LED stripe I turned on sleep mode (1 hour) on my Apple TV which obviously led to this bad response times.
